presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -193,10 +213,41 @@ public synchronized String toString()
                 .toString();
     }
 
-    private synchronized void addRunnerThread()
+    private synchronized void adjustRunnerThreads(int target)
+    {
+        targetRunnerThreads = target;
+
+        int moreThreads = targetRunnerThreads - runnerThreads;
+        if (moreThreads > 0) {
+            addRunnerThreads(moreThreads);
+            runnerThreads = targetRunnerThreads;
+        }
+    }
+
+    private synchronized boolean tryStopRunner()
+    {
+        if (runnerThreads > targetRunnerThreads) {
+            runnerThreads--;
```

Comment:
Maybe. This is just a suggestion of naming which seems more readable to me. Not feeling strongly about it. Do as you wish.